### DISCUSSION TITLE
Allow var for field declarations

### DISCUSSION THREAD

**Discussion by ekolis** [dotnet/csharplang#2828] [Language Ideas]
Right now you can't declare a field like so:

```
var answer = 42;
```

Even if you initialize it with a value.

You have to use the explicit type:

```
int answer = 42;
```

This seems like a silly restriction...

**Comment by HaloFour** [comment]
The "value" could be an expression of arbitrary complexity that could reference (and influence inference) of other fields.  Field initializers are permitted to forward reference unlike locals which can create circular inference situations.

https://github.com/dotnet/csharplang/issues/601
https://github.com/dotnet/roslyn/issues/11551
https://blogs.msdn.microsoft.com/ericlippert/2009/01/26/why-no-var-on-fields/

**Comment by LeaFrock** [comment]
A silly restriction? Are you serious? Please tell me when you see the code `var answer = 42`, the type of 'answer' is Int32, Int64, Double or Decimal? How could the compiler know which type you want the value to be when you only give a 'var'? I think JavaScript may be more useful and suitable to you.

**Reply by joserubenh** [reply]
var a = 23; (int32)
var b = 23m; (decimal)
var c = 56f; (float)

**Reply by joserubenh** [reply]
The compiler is actually very strict about that specific issue, since c++ days.

**Comment by YairHalberstadt** [comment]
@LeaFrock

You can already use `var answer = 42` in methods. `var` is an int in that case.

**Comment by Scott-Caldwell** [comment]
> `var answer = 42`

This is always unambiguously an int (Int32), and my impression is that this is fairly intuitive to most people.

The issue I take with `var` fields is that the type of a field to me is part of the class's contract, and it feels like field types ought to be obvious and explicit. Additionally, since target-typed `new` is likely to be implemented, I feel that that will help to alleviate the concerns around having to explicitly state field types.

**Comment by LeaFrock** [comment]
@YairHalberstadt  
Well, it becomes a conventional thing. But I don't think it's a good use ......while 'var' appears everywhere. Just a view on my own.

>the type of a field to me is part of the class's contract, and it feels like field types ought to be 
obvious and explicit.

Agreed.@Scott-Caldwell
